lib/models/LifecycleConfiguration.js

```diff
@@ -253,9 +253,7 @@ class LifecycleConfiguration {
         // if no Rule Prefix or Filter, rulePrefix is empty string
         filterObj.rulePrefix = '';
         if (Array.isArray(filter)) {
-            // if Prefix was included, not Filter, filter will be Prefix array
-            // if more than one Prefix is included, we ignore all but the last
-            filterObj.rulePrefix = filter.pop();
+            filterObj.rulePrefix = filter[0];
```

Comment:
For removing the `.pop`s, are we going to somehow limit the number of Prefixes someone can include in the configuration? Cos the pops were originally included cos there can be a lot of prefixes, but only the last one in the array should be considered was what I found Amazon functionality to be

Reply:
I saw that you can include multiple prefixes, but that they were being filtered out by the SDK so that the XML that we receive only includes one prefix. 

I didn't try directly sending the XML though, for example with `curl` or the likes.

Do you know of another case? This can easily be updated to `filter[filter.length -1]`, if so.